lib/elixir/lib/enum.ex

```diff
@@ -803,11 +803,13 @@ defmodule Enum do
   @spec filter_map(t, (element -> as_boolean(term)),
     (element -> element)) :: list
 
-  def filter_map(enumerable, filter, mapper) when is_list(enumerable) do
+  def filter_map(enumerable, filter, mapper)
+  when is_list(enumerable) and is_function(filter, 1) and is_function(mapper, 1) do
```

Comment:
> me and @lexmag discuss this stuff way more than we should so I definitely know about that :P

@whatyouhide haha :) Thank you, noted on that differences between styling elixir source code and codes from the wild. No problem at all, just wanted you opinion on the subject